tests/test_preprocessing.py

```diff
@@ -0,0 +1,618 @@
+        ("risk_aversion", 10, 15),
+    ],
+)
+def test_reduce_damages(tmp_path, recipe, eta, batchsize):
+    """
+    Test that reduce_damages returns a Zarr file with damages reduced according to the expected file structure
+    """
+    d = tmp_path / "reduction"
+    d.mkdir()
+    dummy_sector1_dir = d / "dummy_sector1"
+    dummy_sector1_dir.mkdir()
+    dummy_socioeconomics_dir = d / "dummy_se"
+    dummy_socioeconomics_dir.mkdir()
+    dummy_soecioeconomics_file = dummy_socioeconomics_dir / "integration_dummy_se.zarr"
+    reduced_damages_out = d / "reduced_damages"
+
+    config_data = dict(
+        paths=dict(reduced_damages_library=str(d / "reduced_damages")),
+        sectors=dict(
+            dummy_sector1=dict(
+                sector_path=str(dummy_sector1_dir / "dummy_sector1.zarr"),
+                delta="delta_dummy1",
+                histclim="histclim_dummy1",
+            ),
+        ),
+    )
+
+    config_in = d / "config.yml"
+
+    with open(config_in, "w") as outfile:
+        yaml.dump(config_data, outfile, default_flow_style=False)
+
+    damages_ds_1 = xr.Dataset(
+        {
+            "delta_dummy1": (
+                ["gcm", "model", "rcp", "ssp", "batch", "region", "year"],
+                np.ones((2, 2, 2, 2, 15, 2, 2)),
+            ),
+            "histclim_dummy1": (
+                ["gcm", "model", "rcp", "ssp", "batch", "region", "year"],
+                np.ones((2, 2, 2, 2, 15, 2, 2)),
+            ),
+        },
+        coords={
+            "gcm": (["gcm"], ["ABCD1", "surrogate_GCM"]),
+            "model": (["model"], ["IIASA GDP", "OECD Env-Growth"]),
+            "rcp": (["rcp"], ["rcp45", "rcp85"]),
+            "ssp": (["ssp"], ["SSP3", "SSP4"]),
+            "batch": (["batch"], np.arange(15)),
+            "region": (["region"], ["ZWE.test_region", "USA.test_region"]),
+            "year": (["year"], [2022, 2023]),
+        },
+    ).chunk(
+        {
+            "batch": batchsize,
+            "ssp": 1,
+            "model": 1,
+            "rcp": 1,
+            "gcm": 1,
+            "year": 10,
+            "region": 24378,
+        }
+    )
+
+    damages_ds_1.to_zarr(
+        dummy_sector1_dir / "dummy_sector1.zarr", consolidated=True, mode="w"
+    )
+
+    dummy_soecioeconomics = xr.Dataset(
+        {
+            "gdp": (["ssp", "region", "model", "year"], np.ones((2, 2, 2, 3))),
+            "gdppc": (["ssp", "region", "model", "year"], np.ones((2, 2, 2, 3))),
+            "pop": (["ssp", "region", "model", "year"], np.ones((2, 2, 2, 3))),
+        },
+        coords={
+            "ssp": (["ssp"], ["SSP3", "SSP4"]),
+            "region": (["region"], ["ZWE.test_region", "USA.test_region"]),
+            "model": (["model"], ["IIASA GDP", "OECD Env-Growth"]),
+            "year": (["year"], [2021, 2022, 2023]),
+        },
+    )
+
+    dummy_soecioeconomics.to_zarr(
+        dummy_soecioeconomics_file, consolidated=True, mode="w"
+    )
+
+    if batchsize != 15:
```

Comment:
is this block actually getting tested?

Reply:
Yup, one of the pytest fixtures makes batchsize equal to 5 which triggers this conditional.

Reply:
Oh jeez I actually checked the fixture and saw only 15s. Obviously I misread it. Thanks!